### ISSUE TITLE
do you need the philips wiz app still ?

### ISSUE THREAD

**Issue by gedw99** [Dadido3/D3iot#1]
thanks for this. Very interesting how you also wrote some light tools.

Do you still need the philips wiz app to take a new light and get it registered ?  I am trying to avoid that layer of "lock in"

**Comment by Dadido3** [comment]
Yes, unfortunately i haven't found any way to connect the light bulb to the wifi in any other way than with the official app. The light bulb doesn't need internet access to function, but it does need to connect to the official app via bluetooth to set up the wifi configuration.

**Comment by gedw99** [comment]
Ah bummer. They always do this. Tracking or one reason .

I just want to write a nice web PWA to talk to my light bulbs. No server just local calls from iPhone safari to the lights . 

Is this viable ? 

i could keep sone mapping table if needed on google drive and expose it

**Comment by gedw99** [comment]
I would just use golang wasm btw so it’s a service worker installed on my iOS device and loading off the google drive

**Comment by Dadido3** [comment]
You would need to try if PWAs on iOS are allowed to directly communicate via UDP sockets, and if go supports that on WASM. Most browser runtimes will not allow that for security reasons.

So you either have to have some sort of server running that you can talk to via HTTP(S) or websockets and that server will control the lights. Or you need to figure out how to send/receive UDP sockets from whatever sandbox your WASM code is running in.

There is a trick to emulate a TCP socket via websockets. Maybe there is something to emulate UDP sockets, too. But i haven't looked into it. That also depends on the browser/runtime, so it may or may not be fixed (fixed as in: made impossible) on webkit.

If it wasn't for this restriction, you would be able to easily control these light bulbs directly from your phone, but i'm afraid that it will not work that easily.

**Comment by gedw99** [comment]
ah did not know it uses UDP to talk to the light. GLad i asked and they probably won't, but easy for me to try.

I can build a basic IOS gui with GIOUI that imports you lib and find out. 
https://gioui.org/doc/install
https://gioui.org/doc/install/ios

**Comment by gedw99** [comment]
I started to use vuejs . It’s nice and simple .. 

The UDP problem wil be solved with a UDP proxy in golang . It exposes a http api so the gui can acess the UDP socket plan .
